storage/database/src/main/java/de/danoeh/antennapod/storage/database/mapper/FeedItemSortQuery.java

```diff
@@ -26,6 +26,10 @@ public static String generateFrom(SortOrder sortOrder) {
                         + PodDBAdapter.KEY_PLAYBACK_COMPLETION_DATE + " " + "DESC";
             case DATE_OLD_NEW:
                 return PodDBAdapter.TABLE_NAME_FEED_ITEMS + "." + PodDBAdapter.KEY_PUBDATE + " " + "ASC";
+            case EPISODE_FILENAME_A_Z:
+                return PodDBAdapter.KEY_FILE_URL + " " + "ASC";
+            case EPISODE_FILENAME_Z_A:
+                return PodDBAdapter.KEY_FILE_URL + " " + "DESC";
             case DATE_NEW_OLD:
             default:
```

Comment:
This default-clause implies that if a sort order is used that does not have a case-clause that will be treated like `TABLE_NAME_FEED_ITEMS`.

Maybe it would be a good idea to remove the default-clause?

In that case the compiler would emit a warning if an enum constant is missing from the switch, which could catch similar errors in the future. (At least I think there would be a warning. Eclipse emits a warning in this case.)

The default case could be handled after the switch. But maybe it would be better to throw an exception instead?

https://stackoverflow.com/a/30944687/452775

Reply:
I think the default to sort by something else is sensible. If an exception would have been thrown instead, the app would have been unusable until an update because users could not change it back